tests/images.rs

```diff
@@ -380,3 +381,56 @@ fn orientdb_exists_database() {
 
     assert!(!exists);
 }
+
+#[test]
+fn kafka_publish_and_read_message() {
+    let free_local_port = free_local_port().unwrap();
+    let docker = clients::Cli::default();
+    let zk_image =
+        images::zookeeper::Zookeeper::default().with_mapped_port((free_local_port, 2181));
+    let zk_node = docker.run(zk_image);
+    let zk_host_port = zk_node.get_host_port(2181).unwrap();
+    let zk_host_address = zk_node.get_container_gateway();
+    let zk_urls = format!("{}:{}", zk_host_address, zk_host_port);
+    let kafka_image = images::kafka::Kafka::default()
+        .with_env_var("KAFKA_CFG_AUTO_CREATE_TOPICS_ENABLE", "true")
+        .with_env_var("ALLOW_ANONYMOUS_LOGIN", "yes")
+        .with_env_var("ALLOW_PLAINTEXT_LISTENER", "yes")
+        .with_env_var("KAFKA_CFG_AUTO_CREATE_TOPICS_ENABLE", "true")
+        .with_env_var("KAFKA_CFG_ZOOKEEPER_CONNECT", &*zk_urls);
+    let kafka_node = docker.run(kafka_image);
+
+    let message = "ThisIsATestMessage";
+    let producer_command = format!(
+        "echo {} | kafka-console-producer.sh --bootstrap-server localhost:9092 --topic myTestTopic",
+        message.clone()
+    );
+    Command::new("docker")
+        .args(&[
+            "exec",
+            "-i",
+            kafka_node.id(),
+            "bash",
+            "-c",
+            &*producer_command,
+        ])
+        .output()
+        .unwrap();
+
+    let consumer_command = "kafka-console-consumer.sh --bootstrap-server localhost:9092 --topic myTestTopic --group daniel --from-beginning --max-messages 1";
+    let consumer = Command::new("docker")
+        .args(&[
+            "exec",
+            "-i",
+            kafka_node.id(),
+            "bash",
+            "-c",
+            &*consumer_command,
+        ])
+        .output()
+        .unwrap();
+    assert_eq!(
+        &*String::from_utf8(consumer.stdout).unwrap().trim_end(),
+        message.clone()
+    );
```

Comment:
Most of the other tests in here utilize some kind of client library to test the container functionality. Ideally, these tests also act as a form of documentation how to best use these images. Shelling out to bash doesn't feel like the way people would usually use this :)